kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/resolved/KSAnnotationResolvedImpl.kt

```diff
@@ -51,7 +52,12 @@ class KSAnnotationResolvedImpl private constructor(
     }
     override val arguments: List<KSValueArgument> by lazy {
         val presentArgs = annotationApplication.arguments.map {
-            KSValueArgumentImpl.getCached(it, this, origin)
+            val argOrigin = if (origin == Origin.SYNTHETIC) {
```

Comment:
So... the comment below should belong here?
```
// Annotations on deeply synthesized members like getter of Java annotation arguments can still be real.
```
It looks like, even though the `origin` is synthetic for annotation arguments, we're checking its containing file one more time as it could be real?

Reply:
Yes. The name and comment are updated to avoid the confusion.